public/app/features/panel/metrics_panel_ctrl.ts

```diff
@@ -141,46 +180,34 @@ class MetricsPanelCtrl extends PanelCtrl {
     this.intervalMs = res.intervalMs;
   }
 
-  issueQueries(datasource) {
+  issueQueries(datasource: DataSourceApi) {
     this.datasource = datasource;
 
-    if (!this.panel.targets || this.panel.targets.length === 0) {
-      return this.$q.when([]);
+    const panel = this.panel as PanelModel;
+    if (!panel.queryRunner) {
+      panel.queryRunner = new PanelQueryRunner();
```

Comment:
I think this does not belong here, would love to see an api like `panel.getQueryRunner()` that would be responsible for instance creation when necessary.

Reply:
that is a great idea, that if statement is used now in 3 places (here, PanelChrome & QueriesTab). Will fix, refactoring this PR now